3DCalendar/CalendarViewController.swift

```diff
@@ -0,0 +1,197 @@
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if collectionView == collectionView {
+            return numDaysPerMonth[currentMonthIndex-1] + firstWeekDayOfMonth - 1
+        } else {
+            return weekdayData.count
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! DateCell
+        cell.frame.size.width = 25.0
+        cell.frame.size.height = 25.0
+        if indexPath.item <= firstWeekDayOfMonth - 2 {
+            cell.isHidden = true
+        } else {
+            let calcDate = indexPath.row-firstWeekDayOfMonth+2
+            cell.isHidden = false
+            cell.label.text="\(calcDate)"
+            if calcDate < todaysDate && currentYear == presentYear && currentMonthIndex == presentMonthIndex {
+                cell.isUserInteractionEnabled = false
+                cell.label.textColor = UIColor.lightGray
+            } else {
+                cell.isUserInteractionEnabled = true
+                cell.label.textColor = UIColor.black
+            }
+        }
+        return cell
+        
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+        return 0.0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: collectionView.bounds.width / 7.0,
+                      height: collectionView.bounds.width / 7.0)
+    }
+    
+}
+
+class DateCell: UICollectionViewCell {
+    
+    let label: UILabel = {
+        let lbl = UILabel()
+        lbl.text = "00"
+        lbl.textAlignment = .center
+        lbl.font = UIFont.systemFont(ofSize: 12)
+        lbl.textColor = UIColor.gray
+        lbl.translatesAutoresizingMaskIntoConstraints = false
+        return lbl
+    }()
+    
+    override init(frame: CGRect) {
+        let frame2 = CGRect(x: 0, y: 0, width: 25, height: 25)
+        super.init(frame: frame2)
+        backgroundColor = UIColor.clear
+        layer.cornerRadius = 5
+        layer.masksToBounds = true
+        
+        setUpViews()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setUpViews() {
+        addSubview(label)
+        label.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        label.leftAnchor.constraint(equalTo: leftAnchor).isActive = true
+        label.rightAnchor.constraint(equalTo: rightAnchor).isActive = true
+        label.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+    }
+}
+
+extension Date {
```

Comment:
After we do our refactor let's decide if we still need any of this. If we do we can move it to a different file.